Tests/SonarQube.Bootstrapper.Tests/ArgumentProcessorTests.cs

```diff
@@ -257,9 +256,72 @@ public void ArgProc_BeginAndEndVerbs()
+            CheckVerbosity(VerbosityCalculator.DefaultLoggingVerbosity, "Settings are case sensitive", "/d:sonar.VERBOSE=true");
+            CheckVerbosity(VerbosityCalculator.DefaultLoggingVerbosity, "Trace and Debug are independent SonarQube verbosity settings. We only track Debug", "/d:sonar.log.level=trace");
+
+            CheckVerbosity(LoggerVerbosity.Debug, "", "/d:sonar.verbose=true");
+            CheckVerbosity(LoggerVerbosity.Debug, "", "/d:sonar.verbose=TRue");
+            CheckVerbosity(LoggerVerbosity.Info, "", "/d:sonar.verbose=false");
+            CheckVerbosity(LoggerVerbosity.Debug, "", "/d:sonar.log.level=debug");
+            CheckVerbosity(LoggerVerbosity.Debug, "", "/d:sonar.log.level=INFO|DEBUG|TRACE");
+            CheckVerbosity(LoggerVerbosity.Debug, "", "/d:sonar.log.level=***debug***");
+            CheckVerbosity(LoggerVerbosity.Debug, "sonar.verbose takes precedence over sonar.log.level", "/d:sonar.verbose=true", "/d:sonar.log.level=INFO");
+            CheckVerbosity(LoggerVerbosity.Info, "sonar.verbose takes precedence over sonar.log.level", "/d:sonar.verbose=FALSE", "/d:sonar.log.level=DEBUG");
+        }
+
+        [TestMethod]
+        public void ArgProc_SonarVerbose_IsBool()
+        {
+            TestLogger logger = new TestLogger();
+
+            IBootstrapperSettings settings = CheckProcessingSucceeds(logger, "/d:sonar.host.url=foo", "begin", "/d:sonar.verbose=yes");
+            Assert.AreEqual(VerbosityCalculator.DefaultLoggingVerbosity, settings.LoggingVerbosity, "Only expecting true or false");
+
+            logger.AssertErrorsLogged(0);
+            logger.AssertSingleWarningExists("yes");
+        }
 
-        #region Checks
+        [TestMethod]
+        public void ArgProc_SonarVerbose_CmdAndFile()
+        {
+            // Arrange
+            string testDir = TestUtils.CreateTestSpecificFolder(this.TestContext, "settings");
+            string fullPropertiesPath = Path.Combine(testDir, "settings.txt");
+            AnalysisProperties properties = new AnalysisProperties();
+            properties.Add(new Property() { Id = SonarProperties.HostUrl, Value = "http://settingsFile" });
+            properties.Add(new Property() { Id = SonarProperties.LogLevel, Value = "INFO|DEBUG" });
+            properties.Save(fullPropertiesPath);
+
+            TestLogger logger = new TestLogger();
+
+            IBootstrapperSettings settings = CheckProcessingSucceeds(logger, "/s: " + fullPropertiesPath);
+            Assert.AreEqual(LoggerVerbosity.Debug, settings.LoggingVerbosity);
+
+            settings = CheckProcessingSucceeds(logger, "/s: " + fullPropertiesPath, "/d:sonar.verbose=false");
+            Assert.AreEqual(LoggerVerbosity.Info, settings.LoggingVerbosity, "sonar.verbose takes precedence");
+        }
+
+        #endregion Tests
+
+            #region Checks
+
+        private static void CheckVerbosity(LoggerVerbosity expectedVerbosity, string errorMessage, params string[] args)
```

Comment:
The test cases would be easier to read if you put the errorMessage first (avoids having a whole series of strings next to each other, particularly since several of the test cases pass empty messages).